File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        int max_len = 0;
        for(char c: alphabet){
            int l = 0, r = 0, replaced_count = 0;
            while(r < s.size()){
                if(s[r] == c) r++;
                else if(replaced_count < k){
                    r++;
                    replaced_count++;
                }
                else{
                    if(s[l] != c )replaced_count--;
                    if(l == r)r++;
                    l++;
                }
                max_len = max(max_len, r-l);
            }
            
        }
        return max_len;
    }
};

File: Data Structures & Algorithms/crawler-log-folder/submission-3.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        steps = 0
        for i in logs:
            if i == '../':
                steps = max(0, steps-1)
            elif i == './':
                pass
            else:
                steps = steps + 1

        return steps
